#!/usr/bin/env python

PACKAGE = "whycon_ros"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("identify",       bool_t,   0, "Whether to identify IDs of tags.", False);
gen.add("circleDiameter", double_t, 0, "Tag's outer diameter [cm]",        40, 0.0);
gen.add("numMarkers",     int_t,    0, "Number of markers to track.",      10, 0, 100);
gen.add("minSize",        int_t,    0, "Min size of tag [px]",             40, 10, 1000);

gen.add("fieldLength", double_t, 0, "Distance of tags in x-axis in custom coordinate system.", 1.0, 0.0);
gen.add("fieldWidth",  double_t, 0, "Distance of tags in y-axis in custom coordinate system.", 1.0, 0.0);

gen.add("initialCircularityTolerance",   double_t,    0, "Initial circularity test tolerance [%].", 100.0, 0.0, 100.0)
gen.add("finalCircularityTolerance",     double_t,    0, "Final circularity test tolerance [%].",   2.0,  0.0, 100.0)

gen.add("areaRatioTolerance",             double_t,    0, "Tolerance of black and white area ratios  [%].", 40.0,  0.0, 200.0)
gen.add("centerDistanceToleranceRatio",   double_t,    0, "Concentricity test ratio  [%].",                 10.0,  0.0, 100.0)
gen.add("centerDistanceToleranceAbs",     double_t,    0, "Concentricity test absolute [px].",              5.0,  0.0, 25.0)

exit(gen.generate(PACKAGE, "whycon_ros", "whycon"))
